doc/grafana.md

```diff
@@ -0,0 +1,62 @@
+# Grafana Open Source
+
+Grafana open source is open source visualization and analytics software. It allows you to query, visualize, alert on, and explore your metrics, logs, and traces no matter where they are stored. It provides you with tools to turn your time-series database (TSDB) data into insightful graphs and visualizations.
+
+## Getting Started
+
+```nix
+# In `perSystem.process-compose.<name>`
+{
+  services.grafana."gf1".enable = true;
+}
+```
+
+{#tips}
+## Tips & Tricks
+
+{#change-database}
+### Changing Grafana database
+
+By default, Grafana stores data in the `sqlite3` [database](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#database). It also supports `mysql` and `postgres`.
+
+To change the database to `postgres`, we can use the following config:
+
+1. Create `postgres` service.
+
+```nix
+services.postgres."pg1" = {
+  enable = true;
+  listen_addresses = "127.0.0.1";
+  port = 5435;
+  initialDatabases = [{
+    name = "grafana-db";
+  }];
+  initialScript.after = ''
+    CREATE USER gfuser with PASSWORD 'gfpassword' SUPERUSER;
+  '';
+};
+```
+
+2. Create `grafana` service, and change its config to use the `postgres` database.
+
+```nix
+services.grafana."gf1" = {
+  enable = true;
+  http_port = 3001;
+  extraConf = {
+    database = {
+      type = "postgres";
+      host = "127.0.0.1:5435";
+      name = "grafana-db";
+      user = "gfuser";
+      password = "gfpassword";
+    };
+  };
+};
+```
+
+3. Add a setting to start `grafana` only after `postgres` is running.
+
+```nix
+settings.processes."gf1".depends_on."pg1".condition = "process_healthy";
+```
```

Comment:
````suggestion
```nix
{
  services.postgres.pg1 = {
    enable = true;
    listen_addresses = “127.0.0.1”;
  };
  services.grafana.gf1 = {
    enable = true;
    extraConf.database = with config.services.postgres.pg1; {
      type = “postgres”;
      host = “${listen_addresses}:${port}”;
    };
  };
  
  settings.processes."gf1".depends_on."pg1".condition = "process_healthy";
}
```
````

It would be nice to keep these tips/suggestions simple and short. 
I have assumed that not specifying name and user for postgres db in grafana will use postgres by default for both, if not we can explicitly mention.

Reply:
`initialDatabases` and `intialScript` can be part of postgres doc (whenever we add it) and if one wants to play around with postgres configs, they can refer from there.

I say this because, we are better off keeping these mini-tutorials short (with default configs) and not overwhelm the reader.

Reply:
sure, by default grafana looks for the user `root` in database. and for the database `grafana`. i have added config these two over your suggestion. please have a look